### PR TITLE
Revised string best practices for concatenation, interpolated strings

### LINKED ISSUE #11429: Interpolation and concatenation too
The default thread UI culture is used for string interpolation and concatention too, and this isn't straightforward to change. For any sort of backend server it arguably makes far more sense just to ensure each thread is set to use InvariantCulture by default.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: dead2df6-1ec5-b55c-89aa-d05a7a75a0eb
* Version Independent ID: fe538551-dcfc-c244-3f16-864d3f6c2a6e
* Content: [Best Practices for Using Strings in .NET](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings#feedback)
* Content Source: [docs/standard/base-types/best-practices-strings.md](https://github.com/dotnet/docs/blob/master/docs/standard/base-types/best-practices-strings.md)
* Product: **dotnet**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

### PR DESCRIPTION
**Pull request by rpetrusha**
## Revised string best practices for concatenation, interpolated strings

Fixes #11429 

Related to dotnet/samples#865

### CODE REVIEW

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build after dotnet/samples#865 is merged.

File: docs/standard/base-types/best-practices-strings.md [hunk 1]
```diff
@@ -342,10 +342,36 @@ In addition, string comparisons using different versions of .NET or using .NET o
...
+## Displaying and Persisting Formatted Data
+
+When you display non-string data such as numbers and dates and times to users, format them by using the user's cultural settings. By default, the following all use the current thread culture in formatting operations:
+
+- Interpolated strings supported by the [C#](../../csharp/language-reference/tokens/interpolated.md) and [Visual Basic](../../visual-basic/programming-guide/language-features/strings/interpolated-strings.md) compilers.
+
+- String concatenation operations that use the [C#](../../csharp/language-reference/operators/addition-operator.md#string-concatenation) or [Visual Basic](../../visual-basic/programming-guide/language-features/operators-and-expressions/concatenation-operators.md ) or that call the <xref:System.String.Concat%2A?displayProperty=nameWithType> method directly.
```
Comment (BillWagner) [inline docs/standard/base-types/best-practices-strings.md]: I think you accidentally deleted a couple words:

```suggestion
- String concatenation operations that use the [C#](../../csharp/language-reference/operators/addition-operator.md#string-concatenation) or [Visual Basic](../../visual-basic/programming-guide/language-features/operators-and-expressions/concatenation-operators.md ) concatenation operators or that call the <xref:System.String.Concat%2A?displayProperty=nameWithType> method directly.
```

File: docs/standard/base-types/best-practices-strings.md [hunk 2]
```diff
@@ -342,10 +342,36 @@ In addition, string comparisons using different versions of .NET or using .NET o
...
+
+- For string concatenation, do not allow the compiler to perform any implicit conversions. Instead, perform an explicit conversion by calling a `ToString` overload that has a `provider` parameter. For example, the compiler implicitly uses the current culture when converting a <xref:System.Double> value to a string in the following C# code:
+
+  [!code-csharp[Implicit String Conversion](~/samples/snippets/standard/base-types/string-practices/cs/tostring.cs#1)]
+
+  Instead, you an explicitly specify the culture whose formatting conventions are used in the conversion by calling the <xref:System.Double.ToString(System.IFormatProvider)?displayProperty=nameWithType> method, as the following C# code does:
```
Comment (BillWagner) [inline docs/standard/base-types/best-practices-strings.md]:
```suggestion
  Instead, you can explicitly specify the culture whose formatting conventions are used in the conversion by calling the <xref:System.Double.ToString(System.IFormatProvider)?displayProperty=nameWithType> method, as the following C# code does:
```

**Review by BillWagner — APPROVED** [review]
This LGTM @rpetrusha 

You can :shipit: when you are ready.

**Pull request merged**